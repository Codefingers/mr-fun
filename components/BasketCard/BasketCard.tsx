import React from 'react';

import { Product } from '../../types';

function getCardCost(price, quantity) {
  let cost = price;
  cost = price * quantity;
  return cost;
}

type BasketCardType = {
  item: Product;
  removeFromBasket: (itemId: any) => void;
};

export const BasketCard: React.FC<BasketCardType> = (props) => {
  const {
    item: { id, title, price, quantity },
    removeFromBasket,
  } = props;

  const RemoveButton = () => (
    <button
      style={{
        position: 'absolute',
        right: 0,
        top: '50%',
        transform: 'translateY(-50%)',
      }}
      onClick={() => removeFromBasket({ id })}
    >
      Remove
    </button>
  );

  return (
    <div style={{ paddingTop: 24, position: 'relative' }}>
      <h3>{title}</h3>
      <p>Cost: Â£{getCardCost(price, quantity)}</p>
      <p>Quantity: {quantity}</p>
      <RemoveButton />
    </div>
  );
};

export default BasketCard;
