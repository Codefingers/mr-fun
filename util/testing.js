import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

const Providers = ({ children }) => {
  return (
    <ThemeProvider
      theme={{
        colors: {
          body: '#838fa6',
          heading: '#ebedf2',
          accent: '#BA99FF',
          button: '#FFF',
        },
      }}
    >
      {children}
    </ThemeProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';

// override render method
export { customRender as render };
