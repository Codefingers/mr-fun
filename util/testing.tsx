import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../theme';

export const Providers = ({ children }) => {
  return (
    <ThemeProvider
      theme={theme}
    >
      {children}
    </ThemeProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';

// override render method
export { customRender as render };
