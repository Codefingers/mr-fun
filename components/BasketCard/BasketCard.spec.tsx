import React from 'react';
import { render } from '../../util/testing';

import { BasketCard } from './BasketCard';

describe('<BasketCard />', () => {
  const props = {
    item: {
      id: '2',
      title: 'Party Pass',
      price: 100,
      quantity: 1,
    },
  };

  it('renders', () => {
    const { container } = render(<BasketCard {...props} />);
    expect(container).toMatchSnapshot();
  });
});
