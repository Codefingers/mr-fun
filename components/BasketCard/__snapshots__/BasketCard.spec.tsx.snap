// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BasketCard /> renders 1`] = `
<div>
  <div
    style="padding-top: 24px; position: relative;"
  >
    <h3>
      Party Pass
    </h3>
    <p>
      Cost: Â£
      100
    </p>
    <p>
      Quantity: 
      1
    </p>
    <button
      style="position: absolute; right: 0px; top: 50%; transform: translateY(-50%);"
    >
      Remove
    </button>
  </div>
</div>
`;
